pthread/pthread.c

```diff
@@ -659,20 +908,41 @@ int pthread_cond_timedwait(pthread_cond_t *restrict cond,
 		time_t now_us = timespec_to_us(&now);
 		const time_t abstime_us = timespec_to_us(abstime);
 		time_t timeout_us = abstime_us - now_us;
-		err = condWait(*cond, *mutex, timeout_us);
 
-		while (err == -EINTR) {
-			err = mutexLock(*mutex);
-			if (err == EOK) {
-				clock_gettime(CLOCK_REALTIME, &now);
-				now_us = timespec_to_us(&now);
+		if (cond->initialized == 0) {
+			mutexLock(pthread_common.mutex_cond_init_lock);
+			if (cond->initialized == 0) {
+				cond->initialized = 1;
+				err = condCreate(&cond->condh);
+			}
+			mutexUnlock(pthread_common.mutex_cond_init_lock);
+		}
 
-				if (now_us >= abstime_us) {
-					err = -ETIMEDOUT;
-				}
-				else {
-					timeout_us = abstime_us - now_us;
-					err = condWait(*cond, *mutex, timeout_us);
+		if (err == EOK && mutex->initialized == 0) {
+			mutexLock(pthread_common.mutex_cond_init_lock);
+			if (mutex->initialized == 0) {
+				mutex->initialized = 1;
+				err = mutexCreate(&mutex->mutexh);
+			}
+			mutexUnlock(pthread_common.mutex_cond_init_lock);
+		}
+
+		if (err == EOK) {
+			err = condWait(cond->condh, mutex->mutexh, timeout_us);
+
+			while (err == -EINTR) {
```

Comment:
I've mislead you - it's handled in the wrapper: https://github.com/phoenix-rtos/libphoenix/blob/6e57d09a03ed8c30af3751abf4c36fb9bc8af59a/sys/threads.c#L34

but, sadly, it's handled very wrong. It returns 0 on EINTR... So I propose two solutions:
- either replace condWait with phCondWait here,
- or fix condWait implementation in scope of this PR (so mutex is taken and -EINTR is returned).

I favour the second option ;)